Apps/Playground/UWP/App.cpp

```diff
@@ -213,26 +213,41 @@ void App::OnPointerMoved(CoreWindow^, PointerEventArgs^ args)
 {
     if (m_nativeInput != nullptr)
     {
-        const auto& point = args->CurrentPoint;
-        const auto& position = point->RawPosition;
-
-        m_nativeInput->MouseMove(static_cast<int>(position.X), static_cast<int>(position.Y));
-
-        /*
-         * Note: Because PointerPressed only first for the first button press (and no additional ones)
-         * and PointerReleased only fires when all buttons have been released, we need an alternative way
-         * to track additional button presses.
-         * (See Definition - PointerPressed: https://docs.microsoft.com/en-us/uwp/api/windows.ui.core.corewindow.pointerpressed?view=winrt-22000)
-         * (See Remarks - PointerReleased: https://docs.microsoft.com/en-us/uwp/api/windows.ui.core.corewindow.pointerreleased?view=winrt-22000)
-         *
-         * In both of the below functions, we have three booleans that track each button's state and then we
-         * run these two functions to update any buttons that we may have missed.
-         */
-
-        if (point->IsInContact)
+        const auto& position = args->CurrentPoint->RawPosition;
+        const auto& deviceType = args->CurrentPoint->PointerDevice->PointerDeviceType;
+        const auto& deviceSlot = args->CurrentPoint->PointerId;
+        const auto& updateKind = args->CurrentPoint->Properties->PointerUpdateKind;
+
+        if (deviceType == Windows::Devices::Input::PointerDeviceType::Mouse)
+        {
+            m_nativeInput->MouseMove(static_cast<int>(position.X), static_cast<int>(position.Y));
+
+            // Check PointerUpdateKind (type of event) and call the corresponding function, if necessary
+            switch (updateKind)
+            {
+                case PointerUpdateKind::LeftButtonPressed:
+                    m_nativeInput->MouseDown(Babylon::Plugins::NativeInput::LEFT_MOUSE_BUTTON_ID, static_cast<int>(position.X), static_cast<int>(position.Y));
+                    break;
+                case PointerUpdateKind::MiddleButtonPressed:
+                    m_nativeInput->MouseDown(Babylon::Plugins::NativeInput::MIDDLE_MOUSE_BUTTON_ID, static_cast<int>(position.X), static_cast<int>(position.Y));
+                    break;
+                case PointerUpdateKind::RightButtonPressed:
+                    m_nativeInput->MouseDown(Babylon::Plugins::NativeInput::RIGHT_MOUSE_BUTTON_ID, static_cast<int>(position.X), static_cast<int>(position.Y));
+                    break;
```

Comment:
It seems like you can factor this code out into a separate function that is called by this and pointer down.